pkg/controller/jobframework/integrationmanager.go

```diff
@@ -288,6 +288,12 @@ func GetIntegrationsList() []string {
 	return manager.getList()
 }
 
+// GetEnabledIntegrationsListForTest returns the list of currently enabled frameworks.
+func GetEnabledIntegrationsListForTest(tb testing.TB) []string {
```

Comment:
Maybe we don't need this function check at all. Can we instead verify the JobSet CRD is not installed, and then installed.

Reply:
But how it can help? We need to understand is it enabled before and after. CRDs exists or not doesn't help in this case.

Reply:
Good question. Ideally integration tests should assert on user-observable features of the system, rather than some internal variables. Unit tests are meant for internal state.

We already assert it is enabled after install, because we check the effect - workload gets created and webhook sets suspend=true.
For checking prior to install. IIUC the check that there is no JobSet CRD would indicate the integration is disabled.

Reply:
What we want to cover is that after installing CRD Kueue picks it up. 

Testing what happens prior to installation of the CRD is just a sanity check to make sure it wasn't already installed.

Reply:
Make sense. Only one thing that user can also remove CRDs after enable. Maybe we should think how to disable integration on this case.

Reply:
Yeah, this is not supported currently. I think we could do it as part of https://github.com/kubernetes-sigs/kueue/issues/2837. If we track the CRDs with a reconciler, we could react to the event of the CRD being remvoved and unload the controller and the webhook.

Reply:
Ah I see. I missed this issue. Thanks!